### DISCUSSION TITLE
Use package with resource collection paginated

### DISCUSSION THREAD

**Discussion by vanthao03596** [MarcinOrlowski/laravel-api-response-builder#200] [Q&A]
In config file, i can't find class handler ResourceCollection paginate. Can some one help me to use with
ResourceCollection paginate

**Comment by MarcinOrlowski** [comment]
Depends if you want your response to contain paginated data or not. If you want to return the whole record set, then converter should be pretty easy to write. But If you want to include just a certain portion of the paginated data then you will have to do that "by hand" as there's currently now way to pass any argument to the converter so you cannot tell you want i.e. rows from X to Y only. In such case you need to get desired data range and either paginate/convert yourself, or pass result collection to RB for the conversion.

**Reply by vanthao03596** [reply]
I want to include all paginated resource with meta, link, total,.. Can you guide me more detail @MarcinOrlowski ?

**Reply by MarcinOrlowski** [reply]
As already mentioned. Converter cannot do that, because paginated collection does not contain all the data. Theoretically you could use i.e. `$path` or `$query` to get all the rest of your dataset but this would be completely wrong idea from the architecture view point to do this in the converter code.

So to sum up - you need to build the collection with all the data yourself first and then pass that ndw complete collection to `RB` instead.

**Reply by MarcinOrlowski** [reply]
Does the above answer your question?

**Reply by vanthao03596** [reply]
Thank for reply! 
Can you guide me more detail. I'm don't have too much experience with laravel. 
Many thanks @MarcinOrlowski
